src/fabric.c

```diff
@@ -437,6 +447,15 @@ static void ofi_register_provider(struct fi_provider *provider, void *dlhandle)
 	if (ofi_apply_filter(&prov_log_filter, provider->name))
 		ctx->disable_logging = 1;
 
+	/*
+	 * Prevent utility providers from layering on these core providers
+	 * unless explicitly requested.
+	 */
+	if (!strcasecmp(provider->name, "sockets") ||
+	    !strcasecmp(provider->name, "shm") ||
+	    !strcasecmp(provider->name, "efa"))
+		ctx->disable_layering = 1;
```

Comment:
I would expand this check with:
```
    !strcasecmp(provider->name, "efa") ||
    ofi_is_util_prov(provider))
```
So that utility providers are also marked to prevent layering over them.  I don't think this affects the code anywhere else however.